lxd/db/images.go

```diff
@@ -912,6 +912,22 @@ func (c *Cluster) ImagesGetByNodeID(id int64) ([]string, error) {
 	return addresses, err
 }
 
+// ImagesGetProjectsWithImage returns all projects name that the image associated with the given fingerprint.
+func (c *Cluster) ImagesGetProjectsWithImage(fingerprint string) ([]string, error) {
```

Comment:
Why the new function and repetitive calls?

Couldn't you modify your ImagesGetByNodeID to return a map[string][]string{} that would give you a list of project for a fingerprint? (key is fingerprint, value is list of projects)